staging/src/k8s.io/code-generator/hack/lib/codegen.sh

```diff
@@ -0,0 +1,147 @@
+#!/bin/bash
+
+# Copyright 2017 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+CODEGEN_PKG=./$(dirname "${BASH_SOURCE}")/../..
+
+# build generators on-demand
+CODEGEN_BUILT=false
```

Comment:
Also is it possible not to keep this state in this library?
Couldn't user just call install explicitly outside.